tests/profile/admin/test_profile_indicator_admin.py

```diff
@@ -47,6 +96,117 @@ def test_indicator_queryset(self, mocked_request, indicator, qualitative_indicat
+            == qualitative_indicator
+        )
+
+
+@pytest.mark.django_db
+class TestProfileIndicatorAdminChangeForm:
+    def test_indicator_field_for_existing_object(
+        self,
+        client,
+        superuser,
+        profile_indicator,
+        profile,
+        indicator,
+        qualitative_indicator,
+        private_indicator_profile2,
+        public_indicator_profile2,
+    ):
+        """
+        Test indicator field for existing profile indictaor
+        queryset should include all public variables and private
+        variables of current objects profile
+        """
+        client.force_login(user=superuser)
+        url = reverse(
+            "admin:profile_profileindicator_change", args=(profile_indicator.id,)
+        )
+        res = client.get(url, follow=True)
+        assert res.status_code == 200
+        form = res.context["adminform"].form
+
+        assert profile_indicator.profile == profile
+        queryset = form.fields["indicator"].queryset.order_by("id")
+        assert Indicator.objects.all().count() == 4
+        assert queryset.count() == 3
+        assert list(queryset) == [
+            indicator,
+            qualitative_indicator,
+            public_indicator_profile2,
+        ]
+
+
+@pytest.mark.django_db
+class TestProfileIndicatorAdminFormValidations:
+    def test_validation_for_required_fields(self, client, superuser):
+        client.force_login(user=superuser)
+        url = reverse("admin:profile_profileindicator_add")
+        data = {
+            "metadata-TOTAL_FORMS": 0,
+            "metadata-INITIAL_FORMS": 0,
+            "chart_configuration": "{}",
+        }
+        res = client.post(url, data, follow=True)
+
+        assert res.status_code == 200
+        form = res.context["adminform"].form
+        errors = form.errors
+        error_fields = [
+            "profile",
+            "subcategory",
+            "indicator",
+            "content_type",
+            "choropleth_method",
+        ]
+        assert list(errors.keys()) == error_fields
+
+        for field in error_fields:
+            assert errors[field].data[0].message == "This field is required."
+
+    def test_adding_non_existing_indicator(self, client, superuser):
+        client.force_login(user=superuser)
+        url = reverse("admin:profile_profileindicator_add")
+        data = {
+            "metadata-TOTAL_FORMS": 0,
+            "metadata-INITIAL_FORMS": 0,
+            "chart_configuration": "{}",
+            "indicator": 1000000,
+        }
+        res = client.post(url, data, follow=True)
+
+        assert res.status_code == 200
+        form = res.context["adminform"].form
+        errors = form.errors
+        assert (
+            errors["indicator"].data[0].message
+            == "Select a valid choice. That choice is not one of the available choices."
+        )
+
+    def test_adding_private_indicator_from_different_profile(
```

Comment:
I think you have covered all cases where a private variable should not be accessible, what about a test case where it should be accessible?

Reply:
added 2 more tests to cover accesibility of the indicators 👍